tools/shared/src/main/scala/org/scalajs/core/tools/javascript/JSDesugaring.scala

```diff
@@ -420,6 +394,60 @@ object JSDesugaring {
         case Debugger() =>
           js.Debugger()
 
+        case JSSuperConstructorCall(args) =>
+          assert(!containsAnySpread(args),
+              s"Implementation restriction at $pos: cannot call super JS " +
+              "constructor with spread arguments.")
+
+          unnest(args) { (newArgs, env0) =>
+            implicit val env = env0
+
+            val linkedClass = classEmitter.linkedClassByName(enclosingClassName)
+
+            val superCtorCall = {
+              val superCtor = genRawJSClassConstructor(
+                  getSuperClassOfJSClass(linkedClass))
+              val transformedArgs = args.map(transformExpr)
+
+              outputMode match {
+                case OutputMode.ECMAScript51Global | OutputMode.ECMAScript51Isolated =>
+                  js.Apply(
+                      js.BracketSelect(superCtor, js.StringLiteral("call")),
+                      js.This() :: transformedArgs)
+
+                case OutputMode.ECMAScript6 | OutputMode.ECMAScript6StrongMode =>
+                  js.Apply(js.Super(), transformedArgs)
+              }
+            }
+
+            val fieldDefs = for {
+              field @ FieldDef(name, ftpe, mutable) <- linkedClass.fields
+            } yield {
+              implicit val pos = field.pos
+              val defineProperties = {
+                js.BracketSelect(js.BracketSelect(
+                    envField("g"),
+                    js.StringLiteral("Object")),
+                    js.StringLiteral("defineProperties"))
+              }
+              val transformedName = name match {
+                case name: Ident      => transformIdent(name)
+                case StringLiteral(s) => js.StringLiteral(s)
+              }
+              val descriptor = js.ObjectConstr(List(
+                  js.StringLiteral("configurable") -> js.BooleanLiteral(true),
+                  js.StringLiteral("enumerable") -> js.BooleanLiteral(true),
+                  js.StringLiteral("writable") -> js.BooleanLiteral(true),
+                  js.StringLiteral("value") -> transformExpr(zeroOf(ftpe))
+              ))
+              val descriptors = js.ObjectConstr(List(
+                  transformedName -> descriptor))
+              js.Apply(defineProperties, List(js.This(), descriptors))
+            }
```

Comment:
Simple assigns don't allow a `val` (field) to override an inherited *def* (getter).

No batch because, in theory, es 5 does not specify that the order of the fields is kept when iterating over them, which means that the fields could be created in different orders on different objects, which is bad for performance.